server/protocols/webadmin/src/main/java/org/apache/james/webadmin/service/UserMailboxesService.java

```diff
@@ -0,0 +1,135 @@
+        usernamePreconditions(username);
+        Preconditions.checkArgument(!Strings.isNullOrEmpty(mailboxName));
+        MailboxSession mailboxSession = mailboxManager.createSystemSession(USER_NAME, LOGGER);
+        try {
+            mailboxManager.createMailbox(
+                convertToMailboxPath(username, mailboxName, mailboxSession),
+                mailboxSession);
+        } catch (MailboxExistsException e) {
+            LOGGER.info("Attempt to create mailbox {} for user {} that already exists", mailboxName, username);
+        }
+    }
+
+    public void deleteMailboxes(String username) throws MailboxException, UsersRepositoryException {
+        usernamePreconditions(username);
+        MailboxSession mailboxSession = mailboxManager.createSystemSession(USER_NAME, LOGGER);
+        listUserMailboxes(username, mailboxSession)
+            .map(MailboxMetaData::getPath)
+            .forEach(Throwing.consumer(mailboxPath -> deleteMailbox(mailboxSession, mailboxPath)));
+    }
+
+    public List<MailboxResponse> listMailboxes(String username) throws MailboxException, UsersRepositoryException {
+        usernamePreconditions(username);
+        MailboxSession mailboxSession = mailboxManager.createSystemSession(USER_NAME, LOGGER);
+        return listUserMailboxes(username, mailboxSession)
+            .map(mailboxMetaData -> new MailboxResponse(mailboxMetaData.getPath().getName()))
+            .collect(ImmutableCollectors.toImmutableList());
+    }
+
+    public boolean testMailboxExists(String username, String mailboxName) throws MailboxException, UsersRepositoryException {
+        usernamePreconditions(username);
+        Preconditions.checkArgument(!Strings.isNullOrEmpty(mailboxName));
+        MailboxSession mailboxSession = mailboxManager.createSystemSession(USER_NAME, LOGGER);
+        return mailboxManager.mailboxExists(
+            convertToMailboxPath(username, mailboxName, mailboxSession),
+            mailboxSession);
+    }
+
+    public void deleteMailbox(String username, String mailboxName) throws MailboxException, UsersRepositoryException {
+        usernamePreconditions(username);
+        Preconditions.checkArgument(!Strings.isNullOrEmpty(mailboxName));
+        MailboxSession mailboxSession = mailboxManager.createSystemSession(USER_NAME, LOGGER);
+        deleteMailbox(mailboxSession, convertToMailboxPath(username, mailboxName, mailboxSession));
+    }
+
+    private void deleteMailbox(MailboxSession mailboxSession, MailboxPath mailboxPath) throws MailboxException {
+        try {
+            mailboxManager.deleteMailbox(mailboxPath, mailboxSession);
+        } catch (MailboxNotFoundException e) {
+            LOGGER.info("Attempt to delete mailbox {} for user {} that does not exists", mailboxPath.getName(), mailboxPath.getUser());
+        }
+    }
+
+    private void usernamePreconditions(String username) throws UsersRepositoryException {
```

Comment:
Yes, at least for mailbox creation.
Otherwise we create mailboxes that are not linked to users.